### ISSUE TITLE
Installation fails due to conflicting numpy version

### ISSUE THREAD

**Issue by NeolithEra** [tensorlayer/hyperpose#212]
Hi, users are unable to run **openpose-plus** due to dependency conflict with _**numpy**_ package. 
As shown in the following full dependency graph of **_openpose-plus_**, **_openpose-plus_** requires _**numpy>=1.14,<1.16**_，while **_tensorlayer_** requires _**numpy>=1.16,<1.17**_.

According to pip’s “first found wins” installation strategy, _**numpy==1.16.0rc2**_(the latest verison) is the actually installed version. 
However, _**numpy==1.16.0rc2**_ does not satisfy _**numpy>=1.16,<1.17**_.

![333](https://user-images.githubusercontent.com/52778917/62124053-db762800-b2fb-11e9-8e12-e22137644b93.jpg)

### Dependency tree-----
```
openpose-plus(version range:)
| +-cython(version range:>=0.29)
| +-easydict(version range:>=1.8,<=1.10)
| +-nltk(version range:<3.4,>=3.3)
| +-numpy(version range:>=1.14,<1.16)
| +-opencv-python(version range:<3.5,>=3.4)
| +-tensorflow(version range:>=1.8.0,<=1.12.0)
| +-tensorlayer(version range:>=1.11.1)
| | +-imageio(version range:==2.5.0)
| | +-numpy(version range:>=1.16,<1.17)
| | +-progressbar2(version range:==3.39.3)
| | +-requests(version range:==2.21.0)
| | +-scikit-learn(version range:==0.21.0)
| | +-scikit-image(version range:==0.15.0)
| | +-scipy(version range:==1.2.1)
| | +-wrapt(version range:==1.11.1)
| | +-h5py(version range:>=2.9)
| | | +-numpy(version range:>=1.7)
| | | +-six(version range:)
| | +-cloudpickle(version range:>=0.8.1)
```

Thanks for your help.
Best,
Neolith

**Comment by NeolithEra** [comment]
### Solution

1.	Fix your direct dependency to be **_numpy>=1.16,<1.17_** and tensorlayer==2.1.0. 
I have checked this revision will not affect your downstream projects now.

2.	Lock tensorlayer==1.11.1, since tensorlayer 1.11.1 requires numpy>=1.14,<1.16，which can remove the conflict.

Which solution do you prefer, 1 or 2?
Please let me know your choice. I can submit a PR to solve this issue.

**Comment by lgarithm** [comment]
I think 2 would be better, since Tensorlayer has changed a lot in 2.0.
